### PR TITLE
Fix DCIM Shadowed Filter Fields

### PR DESCRIPTION
**Pull request by bryanculver**
# Closes: #2804
# What's Changed

Amalgamation of @HanlinMiao's PRs towards #2804.

This will be retarged towards `next` once PRs start getting merged.

# TODO

- [x] #2845
- [x] #2846
- [x] #2847
- [x] #2881
- [x] #2882 
- [x] #2917

### CODE REVIEW

File: nautobot/dcim/models/racks.py [hunk 1]
```diff
@@ -314,9 +314,7 @@ class Rack(PrimaryModel, StatusModel):
         "outer_depth",
         "outer_unit",
     ]
-    dynamic_group_filter_fields = {
-        "group": "group_id",  # Duplicate filter fields that will be collapsed in 2.0
-    }
```
Comment (HanlinMiao) [inline nautobot/dcim/models/racks.py:319]: If Im reading this and the DynamicGroup code correctly, after eliminating `group_id` from `RackFilterSet` and `RackFilterForm`, we no longer need this line?

**Review by bryanculver — COMMENTED** [review]
:shipit:

File: nautobot/dcim/forms.py [hunk 2]
```diff
@@ -2257,6 +2257,11 @@ class DeviceFilterForm(
         label="Has a primary IP",
         widget=StaticSelect2(choices=BOOLEAN_WITH_BLANK_CHOICES),
     )
+    has_interfaces = forms.NullBooleanField(
+        required=False,
+        label="Has interfaces",
+        widget=StaticSelect2(choices=BOOLEAN_WITH_BLANK_CHOICES),
+    )
```
Comment (glennmatthews) [inline nautobot/dcim/forms.py:2264]: Why add this one specifically to the FilterForm?

File: nautobot/dcim/forms.py [hunk 3]
```diff
@@ -2257,6 +2257,11 @@ class DeviceFilterForm(
...
     virtual_chassis_member = forms.NullBooleanField(
         required=False,
         label="Virtual chassis member",
```
Comment (glennmatthews) [inline nautobot/dcim/forms.py:2267]: Don't the `console_ports`, `console_server_ports`, etc. NullBooleanFields need to be renamed to `has_*` to continue working?

File: nautobot/docs/installation/upgrading-from-nautobot-v1.md [hunk 4]
```diff
@@ -1 +1,141 @@
 # Upgrading from Nautobot v1.X
+
+## Updating from Nautobot 1.5 to 2.0
```
Comment (glennmatthews) [inline nautobot/docs/installation/upgrading-from-nautobot-v1.md:3]: Heading seems a little redundant to me. Perhaps instead:

```suggestion
## UI and REST API Filter Changes
```

File: nautobot/docs/installation/upgrading-from-nautobot-v1.md [hunk 5]
```diff
@@ -1 +1,141 @@
...
+| ConsoleServerPort     | `device`             | Enhanced to support primary key UUIDs in addition to names | `/dcim/console-server-ports/?device=<uuid|name>`|
+|                       | `cabled`             | Renamed to `has_cable`                                     | `/dcim/console-server-ports/?has_cable=True|False`|
+| Device                | `manufacturer`       | Enhanced to support primary key UUIDs in addition to slugs | `/dcim/devices/?manufacturer=<uuid|slug>`|
+|                       | `device_type_id`     | Renamed to `device_type` and enhanced to support primary key UUIDs in addition to slugs | `/dcim/devices/?device_type=<uuid|slug>`|
+|                       | `role`               | Enhanced to support primary key UUIDs in addition to slugs | `/dcim/devices/?role=<uuid|slug>`|
+|                       | `platform`           | Enhanced to support primary key UUIDs in addition to slugs | `/dcim/devices/?platform=<uuid|slug>`|
+|                       | `rack_group_id`      | Renamed to `rack_group`                                    | `/dcim/devices/?rack_group=<uuid|slug>`|
```
Comment (glennmatthews) [inline nautobot/docs/installation/upgrading-from-nautobot-v1.md:19]:
```suggestion
|                       | `rack_group_id`      | Renamed to `rack_group` and enhanced to support primary key UUIDs in addition to slugs | `/dcim/devices/?rack_group=<uuid|slug>`|
```

File: nautobot/docs/installation/upgrading-from-nautobot-v1.md [hunk 6]
```diff
@@ -1 +1,141 @@
...
+## Updating from Nautobot 1.5 to 2.0
+
+### Enhanced Filter Fields
+
+Below is a table documenting [enhanced filter field changes](../release-notes/version-2.0.md#enhanced-filter-fields-2804) in v2.x.
+
+| Model                 | Enhanced Filter Field| Changes                                                  | UI and Rest API endpoints Available in v2.X|
+|-----------------------|----------------------|------------------------------------------------------------|----------------------------------------------|
+| ConsolePort           | `device`             | Enhanced to support primary key UUIDs in addition to names | `/dcim/console-ports/?device=<uuid|name>`|
+|                       | `cabled`             | Renamed to `has_cable`                                     | `/dcim/console-ports/?has_cable=True|False`|
```
Comment (glennmatthews) [inline nautobot/docs/installation/upgrading-from-nautobot-v1.md:12]: I wonder if we should split 'renamed' fields into a separate table from 'enhanced' fields?

File: nautobot/docs/release-notes/version-2.0.md [hunk 7]
```diff
@@ -12,6 +12,54 @@ If you are a user migrating from Nautobot v1.X, please refer to the ["Upgrading
 
 ### Changed
 
+#### Enhanced Filter Fields ([#2804](https://github.com/nautobot/nautobot/pull/2804))
+
+In nautobot v2.X, we changed some of the filter fields to enable filtering by both slugs and UUID primary keys:
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-2.0.md:17]:
```suggestion
Many filter fields have been enhanced to enable filtering by both slugs and UUID primary keys.
```

File: nautobot/docs/installation/upgrading-from-nautobot-v1.md [hunk 8]
```diff
@@ -73,69 +83,70 @@ Below is a table documenting [corrected filter field changes](../release-notes/v
...
-|                       | `site_id`           | instead of `/dcim/virtual-chassis/?site_id=<uuid>`, use `site=<uuid>`                  |
-|                       | `master_id`         | instead of `/dcim/virtual-chassis/?master_id=<uuid>`, use `master=<uuid>`              |
-|                       | `tenant_id`         | instead of `/dcim/virtual-chassis/?tenant_id=<uuid>`, use `tenant=<uuid>`              |
-| VLANGroup             | `region_id`         | instead of `/ipam/vlan-groups/?region_id=<uuid>`, use `region=<uuid>`                  |
-|                       | `site_id`           | instead of `/ipam/vlan-groups/?site_id=<uuid>`, use `site=<uuid>`                      |
-| VLAN                  | `region_id`         | instead of `/ipam/vlans/?region_id=<uuid>`, use `region=<uuid>`                        |
-|                       | `site_id`           | instead of `/ipam/vlans/?site_id=<uuid>`, use `site=<uuid>`                            |
+|-----------------------|---------------------|------------------------------------------------------------------------------------------|
```
Comment (HanlinMiao) [inline nautobot/docs/installation/upgrading-from-nautobot-v1.md:86]: I didnt really change this part, just added spaces at the end of the last column

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**